### PR TITLE
Fix to calculateVlasovGridCoupling. Should now work for all correct g…

### PR DESCRIPTION
**Pull request by ykempf**
…rid geometries.

### CODE REVIEW

**Comment by ykempf** [comment]
Wait for my next pushes. Skipping the projection works as intended and I might have the fix for stitching coming up.

**Comment by ykempf** [comment]
Stitching is still under exploration.

File: sysboundary/ionosphere.cpp [hunk 1]
```diff
@@ -1288,68 +1288,105 @@ namespace SBC {
...
-
+      // Determine the nearest ionosphere node to this point.
+      uint32_t nearestNode = findNodeAtCoordinates(x);
+      int32_t elementIndex = nodes[nearestNode].touchingElements[0];
+      int32_t oldElementIndex;
+      
+      std::unordered_set<int32_t> elementHistory;
+      bool override=false;
+      
+      for (uint toto=0; toto<15; toto++) {
```
Comment (ursg) [inline sysboundary/ionosphere.cpp:1299]: This hardcoded number 15 is a little weird here.

Also: if the loop has run through, should it actually continue to return an empty coupling, or just couple to whatever the closest node was?

**Review by ursg — APPROVED** [review]
This looks pretty good already. I would actually say we can merge this already, and debug the refinement stitching issue in a separate PR?

File: sysboundary/ionosphere.cpp (hunk 1 above)
Comment (ykempf) [reply sysboundary/ionosphere.cpp:1299]: Could be something like MAX_TOUCHING_ELEMENTS? Eventually it should exit anyway, but as long as I'm debugging I don't put a `while(true)`. Shall I add a warning to make this nicer eventually?

**Pull request merged**